main/com.spotify/index.js

```diff
@@ -87,9 +87,27 @@ module.exports = class SpotifyDevice extends Tp.BaseDevice {
             accept: "application/json",
             useOAuth2: this,
         }).catch((e) => {
-            if (!e.detail)
-                throw e;
-            throw new Error(JSON.parse(e.detail).error.message);
+            if (e.code !== "503" && e.code !== "429") {
+                if (!e.detail)
+                    throw e;
+                throw new Error(JSON.parse(e.detail).error.message);
+            }
+            //rate handling error
+            let temp = this;
+            return new Promise(function(resolve, reject) {
+                setTimeout(function() {
+                    Tp.Helpers.Http.get(url, {
+                        accept: "application/json",
+                        useOAuth2: temp,
+                    }).then((response) => {
+                        resolve(response);
+                    }).catch((e) => {
+                        reject(e);
+                    });
+                }, 2000);
+            }).catch((e) => {
+                throw new Error("spotify is temporarily unavailable");
```

Comment:
Add the right code to this error and error messages to the manifest. Otherwise we will only show a generic error to the user instead (see https://github.com/stanford-oval/genie-toolkit/pull/525).